app/models/user.rb

```diff
@@ -53,7 +53,8 @@ class User < ActiveRecord::Base
 
   scope :admin, -> { where(is_admin: true) }
 
-  validates :email, :password, :password_confirmation, presence: true
+  validates :email, presence: true
+  validates :password, :password_confirmation, presence: true, on: :create
```

Comment:
This separation is needed. Otherwise, 'edit profile' will not work, since both edit account and edit profile are working on the same model (User).